tests/test_item_attributes.py

```diff
@@ -1,17 +1,15 @@
-from scrapy.crawler import CrawlerProcess
-from scrapy.utils.project import get_project_settings
+from scrapy.utils.misc import walk_modules
+from scrapy.utils.spider import iter_spider_classes
 
-from locations.exporters.geojson import find_spider_class
+from locations.settings import SPIDER_MODULES
 
 
 def test_item_attributes_type():
-    settings = get_project_settings()
-    process = CrawlerProcess(settings)
-
-    for name in process.spider_loader.list():
-        spider = find_spider_class(name)
-        item_attributes = getattr(spider, "item_attributes", {})
-        assert isinstance(item_attributes, dict)
-
-        if extras := item_attributes.get("extras"):
-            assert isinstance(extras, dict)
+    for mod in SPIDER_MODULES:
+        for module in walk_modules(mod):
+            for spider_class in iter_spider_classes(module):
```

Comment:
Wrap in a function?

Reply:
Wrapped to a separate function. Successfully tested on a small spider - `dataset_attributes` are applied (as expected).